splunk_connect_for_snmp_poller/manager/poller.py

```diff
@@ -171,15 +190,80 @@ def __start_realtime_scheduler_task(self):
         # For debugging purposes better change it to "one second"
         schedule.every(self._args.realtime_task_frequency).seconds.do(
-            automatic_realtime_task,
+            automatic_realtime_job,
+            self._mongo_walked_hosts_coll,
+            self._args.inventory,
+            self.__get_splunk_indexes(),
+            self._server_config,
+            self._local_snmp_engine,
+        )
+
+        schedule.every(self._args.matching_task_frequency).seconds.do(
+            self.process_unmatched_devices_job,
+            self._server_config,
+        )
+
+        automatic_realtime_job(
             self._mongo_walked_hosts_coll,
             self._args.inventory,
             self.__get_splunk_indexes(),
             self._server_config,
             self._local_snmp_engine,
         )
 
+    def add_device_for_profile_matching(self, device: InventoryRecord):
+        self._lock.acquire()
+        self._unmatched_devices[device.host] = device
+        self._lock.release()
+
+    def process_unmatched_devices_job(self, server_config):
+        job_thread = threading.Thread(
+            target=self.process_unmatched_devices, args=[server_config]
+        )
+        job_thread.start()
+
+    def process_unmatched_devices(self, server_config):
+        if self._unmatched_devices:
+            try:
+                profiles = get_profiles(server_config)
+                self._lock.acquire()
+                processed_devices = set()
+                for host, device in self._unmatched_devices.items():
+                    realtime_collection = (
+                        self._mongo_walked_hosts_coll.real_time_data_for(
+                            return_database_id(host)
+                        )
+                    )
+                    if realtime_collection:
+                        descr = extract_desc(realtime_collection)
+
+                        if descr:
+                            assigned_profiles = assign_profiles_to_device(
+                                profiles["profiles"], descr
+                            )
+                            processed_devices.add(host)
+
+                            for profile, frequency in assigned_profiles:
+                                entry_key = create_poller_scheduler_entry_key(
+                                    host, profile
+                                )
+                                new_record = InventoryRecord(
+                                    host,
+                                    device.version,
+                                    device.community,
+                                    profile,
+                                    frequency,
+                                )
+                                self.process_new_job(entry_key, new_record, profiles)
+                for d in processed_devices:
+                    self._unmatched_devices.pop(d)
+            except Exception:
+                traceback.print_exc()
```

Comment:
we can make it logger.exception() it will add traceback and we will be able to pass a message with info